keybot/main.go

```diff
@@ -145,29 +180,36 @@ func addCommands(bot *slackbot.Bot) {
 
 	bot.AddCommand("toggle-dryrun", slackbot.ToggleDryRunCommand{})
 
-	bot.AddCommand("build", slackbot.FuncCommand{
-		Desc: "Build all the things!",
-		Fn:   kingpinHandler,
-	})
-
-	bot.AddCommand("release", slackbot.FuncCommand{
-		Desc: "Release all the things!",
-		Fn:   kingpinHandler,
-	})
-
-	bot.AddCommand("test", slackbot.FuncCommand{
-		Desc: "Test all the things!",
-		Fn:   kingpinHandler,
-	})
-
-	bot.AddCommand("cancel", slackbot.FuncCommand{
-		Desc: "Cancel all the things!",
-		Fn:   kingpinHandler,
-	})
-
-	bot.AddCommand("restart", restartCommand())
-
-	bot.AddCommand("date", slackbot.NewExecCommand("/bin/date", nil, true, "Show the current date"))
+	if bot.User() == "tuxbot" {
+		// Tuxbot only knows how to build linux
+		bot.AddCommand("build", slackbot.FuncCommand{
+			Desc: "Build all the things!",
+			Fn:   kingpinTuxbotHandler,
+		})
```

Comment:
The approach here seems to be to put the bot in a different mode depending on the username that the bot finds itself logging in with. I worry that will be very confusing if someone changes the username later, not realizing that it will break what's going on in this function. Having all the commands active all the time would probably make for more useful error messages.